src/Domain/Session/Model/SessionExpiryDate.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace StepupReadId\Domain\Session\Model;
+
+use DateTimeImmutable;
+
+final class SessionExpiryDate
+{
+    /** @var DateTimeImmutable */
+    private $value;
+
+    private function __construct(DateTimeImmutable $expiryDate)
+    {
+        $this->value = $expiryDate;
+    }
+
+    public static function fromISOString(string $expiryDateISO): SessionExpiryDate
+    {
+        return new self(new DateTimeImmutable($expiryDateISO));
```

Comment:
Here you might want to verify the ISO string is valid. For example, by creating an instance in a try catch block. That way you can present a controlled exception message if the string is not valid.

Leaving it like this is also okay with me. I personally like to have controll over the messages ending up in the logs